pkg/sessiontxn/isolation/base.go

```diff
@@ -468,6 +445,159 @@ func (p *baseTxnContextProvider) getSnapshotByTS(snapshotTS uint64) (kv.Snapshot
 	return snapshot, nil
 }
 
+func (p *baseTxnContextProvider) SetOptionsAfterBegin(txn kv.Transaction) {
+	sessVars := p.sctx.GetSessionVars()
+
+	readReplicaType := sessVars.GetReplicaRead()
+	if readReplicaType.IsFollowerRead() {
+		txn.SetOption(kv.ReplicaRead, readReplicaType)
+	}
+
+	if interceptor := temptable.SessionSnapshotInterceptor(p.sctx, p.infoSchema); interceptor != nil {
+		txn.SetOption(kv.SnapInterceptor, interceptor)
```

Comment:
The `txnSetOption` should be deprecated to avoid mis-usages in the future.

Reply:
`txn.SetOption` is widely used in some internal packages, I don't have confidence to deprecate it since it takes too many capabilities.

Reply:
Or is it possible to also check the newly added `isCommitterworking` inside the `tikvTxn.SetOption`?

Reply:
I did it so, but only in test environment, see https://github.com/pingcap/tidb/pull/52304/files#diff-2aa8e8b32ea03a38f0f9a2850ed09a21848ee98686785d3a7dcee9a4d530f8b9R217-R219

Reply:
Why just check in test? I think checking the txn status when `SetOption` is the key to avoid future mistakes.